mri/reconstruct/fourier.py

```diff
@@ -208,3 +215,227 @@ def adj_op(self, x):
+        platform: string, 'cpu', 'multi-cpu' or 'gpu'
+            string indicating which hardware platform will be used to
+            compute the NUFFT
+        Kd: int or tuple
+            int or tuple indicating the size of the frequency grid,
+            for regridding. If int, will be evaluated
+            to (Kd,)*nb_dim of the image
+        Jd: int or tuple
+            Size of the interpolator kernel. If int, will be evaluated
+            to (Jd,)*dims image
+        n_coils: int
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils
+        """
+        if (n_coils < 1) or (type(n_coils) is not int):
+            raise ValueError('The number of coils should be an integer >= 1')
+        self.nb_coils = n_coils
+        self.shape = shape
+        self.platform = platform
+        self.samples = samples * (2 * np.pi)  # Pynufft use samples in
+        # [-pi, pi[ instead of [-0.5, 0.5[
+        self.dim = samples.shape[1]  # number of dimensions of the image
+
+        if type(Kd) == int:
+            self.Kd = (Kd,) * self.dim
+        elif type(Kd) == tuple:
+            self.Kd = Kd
+        elif Kd is None:
+            # Preferential option
+            self.Kd = tuple([2 * ix for ix in shape])
+
+        if type(Jd) == int:
+            self.Jd = (Jd,) * self.dim
+        elif type(Jd) == tuple:
+            self.Jd = Jd
+        elif Jd is None:
+            # Preferential option
+            self.Jd = (5,) * self.dim
+
+        for (i, s) in enumerate(shape):
+            assert (self.shape[i] <= self.Kd[i]), 'size of frequency grid' + \
+                                                  'must be greater or equal ' \
+                                                  'than the image size'
+
+        print('Creating the NUFFT object...')
+        if self.platform == 'cpu':
+            self.nufftObj = NUFFT_cpu()
+            self.nufftObj.plan(om=self.samples,
+                               Nd=self.shape,
+                               Kd=self.Kd,
+                               Jd=self.Jd,
+                               batch=self.nb_coils)
+
+        elif self.platform == 'multi-cpu':
+            warn('Attemping to use OpenCL plateform. Make sure to '
+                 'have  all the dependecies installed')
+            self.nufftObj = NUFFT_hsa(API='ocl',
+                                      platform_number=None,
+                                      device_number=None,
+                                      verbosity=verbosity)
+
+            self.nufftObj.plan(om=self.samples,
+                               Nd=self.shape,
+                               Kd=self.Kd,
+                               Jd=self.Jd,
+                               batch=1,  # self.nb_coils,
+                               ft_axes=tuple(range(samples.shape[1])),
+                               radix=None)
+
+        elif self.platform == 'gpu':
```

Comment:
You should consider excluding the GPU version from coverall. Check out [this](https://coverage.readthedocs.io/en/coverage-4.2/excluding.html) documentation.

Reply:
As discussed, we decided to not do this as the goal is to not have 100 percent coverage but to know what parts arent tested. As these codes will be constantly used going forward, we plan to keep this ON. Maybe in future we could start running even the GPU versions of codes which could bump up the Coverage numbers